### DISCUSSION TITLE
Startcode adaptive mesh leveling

### DISCUSSION THREAD

**Discussion by heidepiek** [mriscoc/Ender3V2S1#1297] [Q&A]
In the wiki these lines are used in the startcode of cura:

 ; Heat up
M104 S{material_print_temperature_layer_0} ; Set Extruder temperature
M140 S{material_bed_temperature_layer_0} ; Set Heat Bed temperature
G28 ; Home all axes
G27 ; Park toolhead
M190 S{material_bed_temperature_layer_0} ; Wait for Heat Bed temperature
M109 S{material_print_temperature_layer_0} ; Wait for Extruder temperature
;
; Reset settings
M220 S100 ;Reset Feedrate
M221 S100 ;Reset Flowrate
;
; Ender Custom Start G-code
G92 E0 ; Reset Extruder
G28O ; Home optionally if steppers were shutdown
M420 S1 ; activate leveling
;
; Adaptive Mesh Leveling (AML)
C29 A ; use AML
;

Why are M420 S1 and C29 A used at the same time?   
Every time I get an error message in octoprint (conection lost) and the message could not activate bedleveling. Now I blocked M420 S1 with ; ( ;M420 S1 ) in the start code. Then it goes well and I don't get the message anymore and octoprint keeps connection.  So I think you have to choose either one. Is this correct? And is it also correct that after the G28 there is a O? (  G28O  )
Sorry guys for asking such questions, do try to educate myself. Trust that what is written in the wiki is good and I can copy that into the start code without question. Did try to uncomment 1 of them now though. But would like confirmation that I did that right.

Harrie

**Comment by mriscoc** [comment]
Don't confuse C29 with G29, C29 A is for AML (adaptive mesh leveling).
M420 S1 needs a valid mesh in memory.
About errors in Octoprint, please read the Wiki.

**Reply by heidepiek** [reply]
But do you need the M420 in your startcode if C29 A is already present in the startcode? That was my question. I just copy and paste the startcode from your wiki in my startcode. And both gcodes ( M420 S1 and C29 A ) are present. I think you have to choose one of them.

**Reply by heidepiek** [reply]
Have read the wiki, and found out that a few things have been changed in the printer settings. I hadn't looked into this yet. Had backed up octoprint, but not all settings probably transferred over. Have now checked ignore warning. Will tonight, when the printer has finished its print job. Test again to see if it is doing well now with both gcodes in the start code. Understand now that C29 A is special for the aml and G29 for a mesh. Am a bit slow on the uptake sometimes. Sorry Miguel. It`s creality`s fault.  Is because my dwin screen broke, and these are no longer produced. You are now stuck with tjc. As a result, I got all these problems. Before this the printer was doing fine. Do know that is some incompetence on my part. But we will get there. Thanks for the help miguel.

Harrie
